src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
 public class EventManager {
-    public void registerListener(EventListener listener) {
+    private final List<EventListener> listeners = new CopyOnWriteArrayList<>();
+    private final ExecutorService executorService = Executors.newCachedThreadPool();
```

Comment:
Executors.newCachedThreadPool() is being used which is suitable for applications that launch many short-lived tasks. However, it can potentially create a large number of threads if used unwisely. Consider using a fixed thread pool or a single thread executor if the application's concurrency level is known and modest.